File: 2020/11/week1/minhyungPark/algorithm/Leetcode1630.kt
/**
 * Leetcode 1630. Arithmetic Subarrays
 * https://leetcode.com/contest/weekly-contest-212/problems/arithmetic-subarrays/
 */

class Leetcode1630 {

    fun checkArithmeticSubarrays(nums: IntArray, l: IntArray, r: IntArray): List<Boolean> {
        val res = mutableListOf<Boolean>()
        for (i in l.indices) {
            val subArray = nums.slice(l[i]..r[i]).sorted()
            res.add(
                subArray.withIndex()
                    .all { it.index ==0 || subArray[1] - subArray[0] == subArray[it.index] - subArray[it.index-1] }
            )
        }
        return res
    }
}

File: 2020/11/week2/1ilsang/algorithm/leetcode1630.js
/**
 * 1630. Arithmetic Subarrays
 * https://leetcode.com/contest/weekly-contest-212/problems/arithmetic-subarrays/
 * @param {number[]} nums
 * @param {number[]} l
 * @param {number[]} r
 * @return {boolean[]}
 */
const checkArithmeticSubarrays = (nums, l, r) =>
  l.map((start, mapIdx) =>
    nums
      .slice(start, r[mapIdx] + 1)
      .sort((a, b) => a - b)
      .every((n, everyIdx, arr, diff = arr[1] - arr[0]) =>
        everyIdx < 2 ? true : n - arr[everyIdx - 1] === diff
      )
  );
